test/v1/models/response/retrieveChargeHistory/RetrieveChargeHistoryResponseSpec.scala

```diff
@@ -0,0 +1,239 @@
+      |      },
+      |      {
+      |         "taxYear":"2019-20",
+      |         "id":"X123456790A",
+      |         "transactionDate":"2019-06-01",
+      |         "type":"Balancing Charge Debit",
+      |         "amount":600.01,
+      |         "reversalDate":"2019-06-07",
+      |         "reversalReason":"Example reason 2"
+      |      }
+      |   ]
+      |}
+    """.stripMargin
+  )
+
+  val desResponseEmptyHistory: JsValue = Json.parse(
+    """
+      |{
+      |   "history": []
+      |}
+    """.stripMargin
+  )
+
+  val mtdResponseEmptyHistory: JsValue = Json.parse(
+    """
+      |{
+      |   "history": []
+      |}
+    """.stripMargin
+  )
+
+  val desResponseEmptyHistoryObject: JsValue = Json.parse(
+    """
+      |{
+      |   "history": [
+      |     {
+      |     }
+      |   ]
+      |}
+    """.stripMargin
+  )
+
+  val desReponseEmpty: JsValue = Json.parse("""{}""")
+
+  val chargeHistoryResponse: ChargeHistory =
+    ChargeHistory(
+      taxYear = Some("2019-20"),
+      id = Some("X123456790A"),
+      transactionDate = Some("2019-06-01"),
+      `type` = Some("Balancing Charge Debit"),
+      amount = Some(600.01),
+      reversalDate = Some("2019-06-05"),
+      reversalReason = Some("Example reason")
+    )
+
+  val chargeHistoryResponse2: ChargeHistory =
+    ChargeHistory(
+      taxYear = Some("2019-20"),
+      id = Some("X123456790A"),
+      transactionDate = Some("2019-06-01"),
+      `type` = Some("Balancing Charge Debit"),
+      amount = Some(600.01),
+      reversalDate = Some("2019-06-07"),
+      reversalReason = Some("Example reason 2")
+    )
+
+  val retrieveChargeHistoryResponse: RetrieveChargeHistoryResponse =
+    RetrieveChargeHistoryResponse(
+      history = Seq(chargeHistoryResponse)
+    )
+
+  val retrieveChargeHistoryResponseMultiple: RetrieveChargeHistoryResponse =
+    RetrieveChargeHistoryResponse(
+      history = Seq(
+        chargeHistoryResponse,
+        chargeHistoryResponse2
+      )
+    )
+
+  val retrieveChargeHistoryResponseEmptyItem: RetrieveChargeHistoryResponse =
+    RetrieveChargeHistoryResponse(
+      history = Seq()
+    )
+
+  "RetrieveChargeHistoryResponse" when {
+    "read from valid JSON" should {
+      "produce the expected RetrieveChargeHistoryResponse object" in {
+        desResponse.as[RetrieveChargeHistoryResponse] shouldBe retrieveChargeHistoryResponse
+      }
+    }
+
+    "read from valid JSON with multiple charge history items" should {
+      "produce the expected RetrieveChargeHistoryResponse object" in {
+        desResponseWithMultipleHHistory.as[RetrieveChargeHistoryResponse] shouldBe retrieveChargeHistoryResponseMultiple
+      }
+    }
+
+    "read from valid JSON with an empty History array" should {
+      "produce the expected RetrieveChargeHistoryResponse object" in {
+        desResponseEmptyHistory.as[RetrieveChargeHistoryResponse] shouldBe RetrieveChargeHistoryResponse(Seq())
```

Comment:
`RetrieveChargeHistoryResponse(Seq())` -> `retrieveChargeHistoryResponseEmptyItem` (makes it clearer to read the tests that the outcome of this test should be the same as the outcome of the test on L208)